src/pages/blog/starlingx-deploying-your-app-3.md

```diff
@@ -0,0 +1,250 @@
+
+It is worth noting that while deploying a containerized app as a StarlingX app has its advantages and is easy to do, transforming a containerized application into a StarlingX app might not be. Therefore, Helm and FluxCD options are more beginner-friendly and are the typical choice for developers and users who don't plan to create an application that will be officially bundled into the StarlingX project.
+ 
+## Prerequisites
+
+This tutorial expects that your application will be containerized and with ready helm-charts. If this is not your case, please read the blog post already mentioned about Helm.
+
+## Integration with StarlingX System
+
+Some of the advantages of installing an app via system are:
+
+* Once the application tar is built and validated, a user can install and use the application with just two commands, without the need for previous Kubernetes knowledge
+* The FluxCD orchestrates the application of helm charts in a specified order, thus guaranteeing the pods are available before continuing to apply the next helm charts. If you did this by hand, you'd need to verify manually that each pod was correctly applied before proceeding.
+* You can do the helm-overrides easily and the system will guarantee the disponibility of the images in other hosts (ex: controller, worker and storage), in case they also need to deploy pods of your application.
+* The ApplicationFramework takes care of the needed validations to guarantee that an application is fully compatible with StarlingX.
+* An application can schedule specific Kubernetes commands to execute even before it is installed via the LifeCycle hooks.
+* My favourite advantage may be that you can upgrade a StarlingX application via system with an updated tar package of that application, and the system will distribute the new version to the other hosts it is associated with.
+
+## Building the app
+
+For transforming the application from its helm-chart to a tar package that can be installed in a StarlingX Cluster, I'll use the [app-gen-tool](https://opendev.org/starlingx/app-gen-tool) terminal application.
+
+First I'll create a python venv and install the app-gen-tool via pip
+
+```
+python3 -m venv venv
+source ./venv/bin/activate
+pip install git+https://opendev.org/starlingx/app-gen-tool.git#subdirectory=./stx-app-generator/stx-app-generator
+```
+
+Next. I'll copy the "helm-chart" folder from the poc-starlingx to the directory I'm currently working on and create an empty file called `app_manifest.yaml`. Here's the folder structure till now: 
```

Comment:
What is poc-starlingx? It is the first time this name is mentioned in this post, maybe when informing about the application that will be used in this demonstration you can inform that poc-starlingx will be the alias of the application

Reply:
good catch!